File: packages/templates/queues/kafka-node/src/queue/kafka.client.ts
/**
 * Apache Kafka Event Streaming Producer & Consumer
 */
export class KafkaEventBus {
  private brokers: string[];

  constructor(brokers = (process.env.KAFKA_BROKERS || 'localhost:9092').split(',')) {
    this.brokers = brokers;
  }

  async emit(topic: string, event: Record<string, unknown>): Promise<void> {
    console.log(`[Kafka] Emitting to topic "${topic}" via [${this.brokers.join(', ')}] ->`, event);
  }

  async subscribe(topic: string, _handler: (event: unknown) => Promise<void>): Promise<void> {
    console.log(`[Kafka] Subscribed to topic "${topic}"`);
  }
}

export const kafkaBus = new KafkaEventBus();

File: packages/templates/queues/kafka-python/app/queue/kafka_client.py
import os
import json

class KafkaClient:
    def __init__(self, bootstrap_servers: str = None):
        self.bootstrap_servers = bootstrap_servers or os.getenv("KAFKA_BROKERS", "localhost:9092")

    def emit(self, topic: str, payload: dict):
        print(f"[Kafka] Emitting event to '{topic}' on {self.bootstrap_servers}: {json.dumps(payload)}")

    def subscribe(self, topic: str, handler):
        print(f"[Kafka] Subscribed to topic '{topic}'")

kafka_client = KafkaClient()
